src/eduid_webapp/signup/helpers.py

```diff
@@ -165,20 +167,28 @@ def complete_registration(signup_user):
             'message': 'user-out-of-sync'
         }
 
-    secret = current_app.config.get('AUTH_SHARED_SECRET')
+    eppn = signup_user.eppn
+    shared_key = current_app.config.get('AUTH_SHARED_SECRET')
+    if not isinstance(shared_key, six.binary_type):
+        shared_key = shared_key.encode('ascii')
     timestamp = '{:x}'.format(int(time.time()))
+    nonce = nacl.utils.random(nacl.secret.SecretBox.NONCE_SIZE)
+    token_data = '{0}|{1}'.format(timestamp, eppn).encode('ascii')
+    box = nacl.secret.SecretBox(shared_key)
+    encrypted = box.encrypt(token_data, nonce)
```

Comment:
same here, don't pass nonce to encrypt

Reply:
It seems that a previous comment I added here got lost. I have now removed the nonce from the encryption, but I'm still sending it in the GET request - so that the same authn app can deal with hmac and secret box tokens. My idea was that, since I was sending the token anyway, I might as well send the actually used one. But since it is not used in the authn app for decryption, it does not really matter to send a dummy nonce.
But why is it not good to use the nonce in the encryption?